src/test/java/org/embulk/filter/column/path/TestPathCompiler.java

```diff
@@ -0,0 +1,106 @@
+package org.embulk.filter.column.path;
+
+import org.embulk.EmbulkTestRuntime;
+import org.embulk.config.ConfigLoader;
+import org.embulk.config.ConfigSource;
+import org.embulk.filter.column.ColumnFilterPlugin;
+import org.embulk.filter.column.ColumnFilterPlugin.ColumnConfig;
+import org.embulk.filter.column.path.ArrayPathToken;
+import org.embulk.filter.column.path.PathCompiler;
+import org.embulk.filter.column.path.PropertyPathToken;
+import org.embulk.spi.Exec;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class TestPathCompiler
+{
+    @Rule
+    public EmbulkTestRuntime runtime = new EmbulkTestRuntime();
+
+    private ConfigSource config()
+    {
+        return runtime.getExec().newConfigSource();
+    }
+
+
+    private ColumnFilterPlugin.PluginTask taskFromYamlString(String... lines)
+    {
+        StringBuilder builder = new StringBuilder();
+        for (String line : lines) {
+            builder.append(line).append("\n");
+        }
+        String yamlString = builder.toString();
+
+        ConfigLoader loader = new ConfigLoader(Exec.getModelManager());
+        ConfigSource config = loader.fromYamlString(yamlString);
+        return config.loadConfig(ColumnFilterPlugin.PluginTask.class);
+    }
+
+    @Test
+    public void isJsonPathNotation_DotNotation()
+    {
+        ColumnFilterPlugin.PluginTask task = taskFromYamlString(
```

Comment:
Do you need this? Creating ColumnConfig lloks enough. 

And, I prefer to seperate PathCompiler with plugin's data structure so that we can release jsonpath compiler seperately in the future.

Having a glue in this plugin namespace should be fine